pyanaconda/modules/timezone/timezone.py

```diff
@@ -0,0 +1,140 @@
+        self._is_utc = Signaled(False)
+        self._ntp_enabled = Signaled(True)
+        self._ntp_servers = Signaled([])
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(TimezoneInterface(self), MODULE_TIMEZONE_PATH)
+        DBus.register_service(MODULE_TIMEZONE_NAME)
+
+    def ping(self, s):
+        """Ping the module."""
+        log.debug(s)
+        return "TimeModule says hi!"
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return TimezoneKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        self.set_timezone(data.timezone.timezone)
+        self.set_utc(data.timezone.isUtc)
+        self.set_ntp(not data.timezone.nontp)
+        self.set_ntp_servers(data.timezone.ntpservers)
+
+    def generate_kickstart(self):
+        """Return the kickstart string."""
+        data = self.get_kickstart_data()
+        data.timezone.timezone = self.timezone
+        data.timezone.isUtc = self.is_utc
+        data.timezone.nontp = not self.ntp_enabled
+
+        if self.ntp_enabled:
+            data.timezone.ntpservers = list(self.ntp_servers)
+
+        return str(data)
+
+    @property
+    def module_changed(self):
+        """Signal that module might have changed.
+
+        If someone changes properties of this module
+        on the python level (and not from DBus), he
+        should emit this signal once he is done.
+        """
+        return self._module_changed
+
+    @property
+    def timezone(self):
+        """Return the timezone."""
+        return self._timezone.value
+
+    @property
+    def timezone_changed(self):
+        """Signal for timezone."""
+        return self._timezone.signal
+
+    def set_timezone(self, timezone):
+        """Set the timezone."""
+        self._timezone.value = timezone
+
+    @property
+    def is_utc(self):
+        """Is the hardware clock set to UTC?"""
+        return self._is_utc.value
+
+    @property
+    def is_utc_changed(self):
+        """Signal for is utc."""
+        return self._is_utc.signal
+
+    def set_utc(self, is_utc):
+        """Set if the hardware clock is set to UTC."""
+        self._is_utc.value = is_utc
+
+    @property
+    def ntp_enabled(self):
+        """Enable automatic starting of NTP service."""
+        return self._ntp_enabled.value
+
+    @property
+    def ntp_enabled_changed(self):
+        """Signal for ntp enabled."""
+        return self._ntp_enabled.signal
+
+    def set_ntp(self, ntp_enabled):
+        """Enable or disable automatic starting of NTP service."""
+        self._ntp_enabled.value = ntp_enabled
+
+    @property
+    def ntp_servers(self):
+        """Return a list of NTP servers."""
+        return self._ntp_servers.value
+
+    @property
+    def ntp_servers_changed(self):
+        """Signal for ntp servers."""
+        return self._ntp_servers.signal
```

Comment:
The Signaled objects should be just an implementation detail hidden from everyone. Then you can provide your own implementation of the DBus interface without the Signaled objects. And when you want to access the properties of the implementation module from python, you don't have to think about signals and Signaled at all.

Reply:
I don't think it should be hidden as implementation detail because it gives you too much overhead in development. I more see it as a data object, holding just one information and giving you info when the value changed.
Of course as I said before, you then need to work with it other way than normal properties which is confusing.